src/compute-engine/boxed-expression/canonical.ts

```diff
@@ -22,6 +22,17 @@ export function canonicalForm(
 
   if (typeof forms === 'string') forms = [forms];
 
+  // Like for full canonicalization, request the canonical form of symbols.
+  // Automatically, this involves the substitution of the symbol with its value, if it is a
+  // constant-flagged symbol, with a 'holdUntil' attribute of 'never'
+  // (@note: the reasoning for carrying this out here is because in some sense, firstly,
+  // 'CanonicalForm' can be regarded as producing a 'canonical' expression, (albeit a 'custom' one;
+  // notably also with the resulting expr. having an 'isCanonical' value of 'true'). Secondly &
+  // following from this, symbol canonicalization (and substitution where appropriate) facilitates
+  // several simplifications which would otherwise not be made: for example 'x^y' where 'y=0', for
+  // canonicalPower.
+  expr = symbolForm(expr);
+
```

Comment:
I have seen that a `Symbol` CanonicalForm has been thought about/mentioned via comments in one or two places, but it seems to me that doing this consistently (for partial canonicalization) makes sense, given the reasoning in this associated comment, and the one prior. What do you think ?

Reply:
Yes, that's reasonable.